4-Beyond-Classical-Search/hillClimbing.js

```diff
@@ -1,22 +1,95 @@
-var HillClimber = function(){
-	// Should the robot go left, right or stay where it is
-	this.DECISIONS = {
-		LEFT:-1,
-		RIGHT:1,
-		STAY:0
-	};
-	// Decide where to go based on height
-	this.decide = function(left,current,right){
-		if(left == right && left > current){
-			this.currentDecision = Math.random() > .5 ?
-			this.DECISIONS.LEFT: this.DECISIONS.RIGHT;
-		} else if(right == null || left > right){
-			this.currentDecision = this.DECISIONS.LEFT;
-		} else if (left == null || right > left) {
-			this.currentDecision = this.DECISIONS.RIGHT;
-		} else {
-			this.currentDecision = this.DECISIONS.STAY;
+class Hill {
+
+	getRandomValues() {
```

Comment:
This is clever! I tried running it several times and each time it gave me a new set of nice hills.